packages/mobile/src/onboarding/registration/JoinCelo.tsx

```diff
@@ -0,0 +1,252 @@
+    pincodeType: state.account.pincodeType,
+    acceptedTerms: state.account.acceptedTerms,
+  }
+}
+
+function getPhoneNumberState(
+  phoneNumber: string,
+  countryCallingCode: string,
+  countryCodeAlpha2: string
+) {
+  const phoneDetails = parsePhoneNumber(phoneNumber, countryCallingCode)
+
+  if (phoneDetails) {
+    return {
+      nationalPhoneNumber: phoneDetails.displayNumber,
+      e164Number: phoneDetails.e164Number,
+      isValidNumber: true,
+      countryCodeAlpha2: phoneDetails.regionCode!,
+    }
+  } else {
+    return {
+      nationalPhoneNumber: phoneNumber,
+      e164Number: '',
+      isValidNumber: false,
+      countryCodeAlpha2,
+    }
+  }
+}
+
+export class JoinCelo extends React.Component<Props, State> {
+  static navigationOptions = nuxNavigationOptions
+
+  countries = new Countries(this.props.i18n.language)
+
+  state: State = {
+    name: this.props.cachedName || '',
+    ...getPhoneNumberState(
+      this.props.cachedNumber || '',
+      this.props.cachedCountryCallingCode || '',
+      RNLocalize.getCountry()
+    ),
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    const prevCountryCodeAlpha2 = prevProps.route.params?.selectedCountryCodeAlpha2
+    const countryCodeAlpha2 = this.props.route.params?.selectedCountryCodeAlpha2
+    if (prevCountryCodeAlpha2 !== countryCodeAlpha2 && countryCodeAlpha2) {
+      const countryCallingCode =
+        this.countries.getCountryByCodeAlpha2(countryCodeAlpha2)?.countryCallingCode ?? ''
+      this.setState((prevState) =>
+        getPhoneNumberState(prevState.nationalPhoneNumber, countryCallingCode, countryCodeAlpha2)
+      )
+    }
+  }
+
+  goToNextScreen = () => {
+    if (!this.props.acceptedTerms) {
+      navigate(Screens.RegulatoryTerms)
+    } else if (this.props.pincodeType === PincodeType.Unset) {
+      navigate(Screens.PincodeEducation)
+    } else {
+      navigate(Screens.EnterInviteCode)
+    }
+  }
+
+  onChangeNameInput = (value: string) => {
+    this.setState({
+      name: value,
+    })
+  }
+
+  onChangePhoneNumberInput = (nationalPhoneNumber: string, countryCallingCode: string) => {
+    this.setState((prevState) =>
+      getPhoneNumberState(nationalPhoneNumber, countryCallingCode, prevState.countryCodeAlpha2)
+    )
+  }
+
+  onPressCountry = () => {
+    navigate(Screens.SelectCountry, {
+      countries: this.countries,
+      selectedCountryCodeAlpha2: this.state.countryCodeAlpha2,
+    })
+  }
+
+  onPressContinue = () => {
+    this.props.hideAlert()
+
+    const { name, e164Number, isValidNumber, countryCodeAlpha2 } = this.state
+    const countryCallingCode = this.countries.getCountryByCodeAlpha2(countryCodeAlpha2)
+      ?.countryCallingCode
+    const { cachedName, cachedNumber, cachedCountryCallingCode } = this.props
+
+    if (
+      cachedName === name &&
+      cachedNumber === e164Number &&
+      cachedCountryCallingCode === countryCallingCode
+    ) {
+      this.goToNextScreen()
```

Comment:
Should we add `this.props.setPromptForno(true)` here as well?

Reply:
It's persisted.